### PR TITLE
[Feature][test] Add MicroBench module to make it easier for developers to write JMH TEST

### LINKED ISSUE #3995: [Feature][test] Add MicroBench module to make it easier for developers to write JMH TEST
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Seatunnel currently lacks benchmarks, so this module needs to be added to solve this problem.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liugddx**
close #3995 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by CalvinKirs — CHANGES_REQUESTED** [review]
Please add the skywalking licenses, I think these code from skywaling

**Review by EricJoy2048 — DISMISSED** [review]

**Comment by liugddx** [comment]
> Please add the skywalking licenses, I think these code from skywaling

Done.

**Comment by CalvinKirs** [comment]
> > Please add the skywalking licenses, I think these code from skywaling↳
> 
> Done.
NOTICE file you have not updated

**Comment by liugddx** [comment]
> > > Please add the skywalking licenses, I think these code from skywaling↳
> > 
> > 
> > Done.
> > NOTICE file you have not updated

Done. PTAL thanks.

File: seatunnel-microbench/src/main/test/org/apache/seatunnel/microbench/core/queue/QueueBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+
+@State(Scope.Thread)
+public class QueueBenchmark extends AbstractMicrobenchmark {
+
```
Comment (CalvinKirs) [inline seatunnel-microbench/src/main/test/org/apache/seatunnel/microbench/core/queue/QueueBenchmark.java]: this is almost nothing to test, and there are even official benchmark tests. If you want to prove that the business of the engine is more suitable for using Disruptor, please bring specific scenarios for use.

Comment (liugddx) [reply seatunnel-microbench/src/main/test/org/apache/seatunnel/microbench/core/queue/QueueBenchmark.java]: > 

Ok. I will provide test cases after the #3847 merge.

**Pull request closed without merging**